modules/core/src/main/java/org/apache/synapse/util/swagger/SchemaValidationUtils.java

```diff
@@ -0,0 +1,272 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class for API schema validation related operations.
+ */
+public class SchemaValidationUtils {
+
+    private static final Log logger = LogFactory.getLog(SchemaValidationUtils.class);
+
+    /**
+     * Utility function to extract collection of String from Map when the key is given.
+     *
+     * @param map  Map of String, Collection<String>
+     * @param name key
+     * @return Collection of Strings
+     */
+    public static Collection<String> getFromMapOrEmptyList(Map<String, Collection<String>> map, String name) {
+
+        if (name != null && map.containsKey(name)) {
+
+            return map.get(name).stream()
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
+        } else {
+            return Collections.emptyList();
+        }
+
+    }
+
+    /**
+     * Utility function to extract query params from resource path.
+     *
+     * @param apiResource resource path with query params.
+     * @return Map of Query params.
+     * @throws UnsupportedEncodingException When failed to decode
+     */
+    public static Map<String, Collection<String>> getQueryParams(String apiResource)
+            throws UnsupportedEncodingException {
+        Map<String, String> queryParams = new HashMap<>();
+        String queryString = apiResource.split("\\?")[1];
+        String[] query = queryString.split("&");
+        for (String keyValue : query) {
+            int idx = keyValue.indexOf("=");
+            queryParams.put(
+                    URLDecoder.decode(keyValue.substring(0, idx), "UTF-8"),
+                    URLDecoder.decode(keyValue.substring(idx + 1), "UTF-8"));
+        }
+        return queryParams.entrySet()
+                .stream()
+                .collect(Collectors.toMap(
+                        Map.Entry::getKey, entry -> Collections.singleton(entry.getValue())));
+    }
+
+    /**
+     * Get the Request/Response messageContent as a JsonObject.
+     *
+     * @param messageContext Message context
+     * @return JsonElement which contains the request/response message content
+     */
+    public static Optional<String> getMessageContent(MessageContext messageContext) {
+
+        Optional<String> payloadObject = Optional.empty();
+        org.apache.axis2.context.MessageContext axis2Context = ((Axis2MessageContext) messageContext)
+                .getAxis2MessageContext();
+        boolean isMessageContextBuilt = isMessageContextBuilt(axis2Context);
+        if (!isMessageContextBuilt) {
+            // Build Axis2 Message.
+            try {
+                RelayUtils.buildMessage(axis2Context);
+            } catch (IOException | XMLStreamException e) {
+                logger.error(" Unable to build axis2 message");
```

Comment:
Should not we throw this exception out?

Reply:
Here, getting the payload is optional ex. in the case of a GET request. 
Also, I belive we should't disrupt the mediation flow until we are sure of a validation error. 

We are reusing the code in [ APIM](https://github.com/wso2/carbon-apimgt/blob/5ee9d1eb160cbf5aaac6757ba58c26e3d46ee2b7/components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/utils/SchemaValidationUtils.java#L115)